assets/js/util/index.js

```diff
@@ -115,73 +116,74 @@ export const removeURLParameter = ( url, parameter ) => {
+	number = isFinite( number ) ? number : Number( number );
 
-	// Handle errors after calculations.
-	if ( isNull( number ) || isUndefined( number ) || isNaN( number ) ) {
-		readableNumber = '';
+	if ( ! isFinite( number ) ) {
+		// eslint-disable-next-line no-console
+		console.warn( 'Invalid number', number, typeof number );
 		number = 0;
 	}
 
-	if ( 0 === number ) {
-		readableNumber = '0.00';
-		return currencyCode
-			? new Intl.NumberFormat( navigator.language, { style: 'currency', currency: currencyCode } ).format( number )
-			: number;
+	if ( currencyCode ) {
+		return numberFormat( number, { style: 'currency', currency: currencyCode } );
 	}
 
-	// Format as amount if currencyCode is passed.
-	if ( false !== currencyCode && '' !== readableNumber ) {
-		const formatedParts = new Intl.NumberFormat( navigator.language, { style: 'currency', currency: currencyCode } ).formatToParts( number );
-
-		const decimal = formatedParts.find( ( part ) => 'decimal' === part.type );
-		if ( ! isUndefined( decimal ) && ! isUndefined( decimal.value ) && 1000 > number ) {
-			readableNumber = Number.isInteger( number ) ? number : number.replace( '.', decimal.value );
-		}
-
-		const currencyFound = formatedParts.find( ( part ) => 'currency' === part.type );
-		const currency = currencyFound ? currencyFound.value : '';
+	const withSingleDecimal = {
+		minimumFractionDigits: 1,
+		maximumFractionDigits: 1,
+	};
 
-		return `${ currency }${ readableNumber }`;
+	switch ( true ) {
+		case 1000000 < number :
+			return sprintf(
+				// translators: %s: an abbreviated number in millions.
+				__( '%sM', 'google-site-kit' ),
+				numberFormat( number / 1000000, withSingleDecimal )
+			);
+		case 99000 < number :
+			return sprintf(
+				// translators: %s: an abbreviated number in thousands.
+				__( '%sK', 'google-site-kit' ),
+				numberFormat( Math.round( number / 1000 ) )
+			);
+		case 1000 < number :
+			return sprintf(
+				// translators: %s: an abbreviated number in thousands.
+				__( '%sK', 'google-site-kit' ),
+				numberFormat( number / 1000, withSingleDecimal )
+			);
+		default:
+			return number.toString();
 	}
-
-	return readableNumber.toString();
 };
 
 /**
  * Internationalization Number Format.
  *
  * @param {number} number The number to format.
- * @param {string} locale Optional, locale to format as amount, default to Browser's locale.
+ * @param {Object} [options] Formatting options.
+ * @param {string} [locale] Locale tag. Optional.
  *
  * @return {string} The formatted number.
  */
-export const numberFormat = ( number, locale = '' ) => {
-	if ( ! locale ) {
-		locale = navigator.language;
-	}
+export const numberFormat = ( number, options = {}, locale = getLocale() ) => {
+	return new Intl.NumberFormat( locale, options ).format( number );
+};
+
+/**
+ * Gets the current locale for use with browser APIs.
+ *
+ * @return {string} Current Site Kit locale if set, otherwise the current language set by the browser.
```

Comment:
Either way would be good to clarify the format here with an example, e.g. `en-us` or something like that.